spec/map_generator/allocated_objects_strategy/object_tracker_spec.rb

```diff
@@ -3,32 +3,42 @@
 require 'spec_helper'
 
 describe Crystalball::MapGenerator::AllocatedObjectsStrategy::ObjectTracker do
-  subject(:tracker) { described_class.new(only_of: only_of) }
-
-  let(:only_of) { ['Object'] }
-  let(:obj1) { double }
-  let(:obj2) { double(class: Dummy) }
+  subject(:tracker) { described_class.new }
 
   before do
     stub_const('Dummy', Class.new)
-    allow(ObjectSpace).to receive(:each_object).with(Object).and_yield(obj1)
+    stub_const('SubDummy', Class.new(Dummy))
   end
 
-  describe '#created_during' do
-    subject do
-      tracker.created_during do
-        allow(ObjectSpace).to receive(:each_object).with(Object).and_yield(obj1).and_yield(obj2)
-      end
+  describe '#used_classes_during' do
+    it 'returns objects allocated during a block' do
+      expect(tracker.used_classes_during { Dummy.allocate }).to match_array(Dummy)
+    end
+
+    it 'returns objects created during a block' do
+      expect(tracker.used_classes_during do
+        Dummy.new
+        Object.new
+      end).to match_array([Object, Dummy])
     end
 
-    it 'lists objects' do
-      expect(subject).to eq [obj2]
+    it 'otherwise' do
```

Comment:
documented output will be
```
Crystalball::MapGenerator::AllocatedObjectsStrategy::ObjectTracker
  #used_classes_during
    otherwise
```
That is not clear when you see it

Reply:
:+1:
`it 'returns empty array when no object is allocated'` for example